invenio_rdm_records/resources/iiif.py

```diff
@@ -393,3 +398,156 @@ def _rewrite_url(self):
+    To ensure permissions are respected the Cantaloupe instance *must* only be accessible by RDM.
+    All IIIF requests from end users are proxied through the app, which enforces access control.
+
+    Attributes:
+        iiif_server (str): URL of IIIF server, <scheme>://<server>[:<port>]/ (required).
+        fs_prefix: (str): Prefix for local filestore (required if Cantaloupe's 'FilesystemSource.BasicLookupStrategy.path_prefix'
+            setting doesn't point directly to the root of the instance's local file bucket)
+        chunk_size (int): Chunk size for streaming file content (default: 64KB).
+        fallback (bool): Fall back to RDM IIIF handling on proxy failure (default: False).
+        iiif_version (int): Major IIIF version to use (default: 2).
+        levels (int): Number of levels in the path RDM uses in local filestore, should be loaded from config.
+        timeout (int): Seconds to wait for Cantaloupe to respond (default: 5).
+    """
+
+    def __init__(self):
+        """Initialize the proxy."""
+        self.iiif_server = current_app.config.get("RDM_IIIF_SERVER_URL")
+        if not self.iiif_server:
+            raise RuntimeError("RDM_IIIF_SERVER_URL not configured")
+        self.fs_prefix = current_app.config.get("RDM_CANTALOUPE_FS_PREFIX", "")
+        self.chunk_size = current_app.config.get("RDM_CANTALOUPE_CHUNK_SIZE", 64 * 1024)
+        self.fallback = current_app.config.get("RDM_IIIF_PROXY_FALLBACK", False)
+        self.iiif_version = current_app.config.get("RDM_CANTALOUPE_IIIF_VERSION", 2)
+        self.levels = current_app.config.get("FILES_REST_STORAGE_PATH_DIMENSIONS", 2)
+        self.timeout = current_app.config.get("RDM_IIIF_PROXY_TIMEOUT", 5)
+        self.iiif_service = current_app.extensions["invenio-rdm-records"].iiif_service
+
+    @property
+    def iiif_image_uuid(self):
+        """'UUID' of the IIIF image (<record_type>:<record_id>:<filename>)."""
+        return resource_requestctx.view_args["uuid"]
+
+    @property
+    def uuid_parts(self):
+        """Parts of the 'UUID' required by the class."""
+        record_type, record_id, filename = self.iiif_image_uuid.split(":")
+        return {
+            "record_type": record_type,
+            "record_id": record_id,
+            "filename": filename,
+            "iiif_record_uuid": ":".join([record_type, record_id]),
+        }
+
+    def file_service(self):
+        """Return the file service for the current record type."""
+        return self.iiif_service.file_service(self.uuid_parts["record_type"])
+
+    def media_files_service(self):
+        """Return the media files service for the current record type."""
+        return self.iiif_service.media_files_service(self.uuid_parts["record_type"])
+
+    def record(self):
+        """Read record via record service."""
+        return self.iiif_service.read_record(
+            g.identity, self.uuid_parts["iiif_record_uuid"]
+        )._record
+
+    def file_id(self):
```

Comment:
```suggestion
    @property
    def file_service(self):
        """Return the file service for the current record type."""
        return self.iiif_service.file_service(self.uuid_parts["record_type"])

    @property
    def media_files_service(self):
        """Return the media files service for the current record type."""
        return self.iiif_service.media_files_service(self.uuid_parts["record_type"])

    @property
    def record(self):
        """Read record via record service."""
        return self.iiif_service.read_record(
            g.identity, self.uuid_parts["iiif_record_uuid"]
        )._record

    @property
    def file_id(self):
```

nit: maybe these should be properties?

Reply:
Fine with me, property usages also updated.